src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy

```diff
@@ -176,29 +208,68 @@ class PalantirDockerPlugin implements Plugin<Project> {
         if (ext.pull) {
             buildCommandLine.add '--pull'
         }
-        buildCommandLine.addAll(['-t', "${ -> ext.name}", '.'])
+        buildCommandLine.addAll(['-t', "${-> ext.name}", '.'])
         return buildCommandLine
     }
 
+    @Deprecated
     private static String computeName(String name, String tag) {
-        int lastColon = name.lastIndexOf(':')
-        int lastSlash = name.lastIndexOf('/')
+        int firstAt = tag.indexOf("@")
 
-        int endIndex;
+        String tagValue
+        if (firstAt > 0) {
+            tagValue = tag.substring(firstAt + 1, tag.length())
+        } else {
+            tagValue = tag
+        }
 
-        // image_name -> this should remain
-        // host:port/image_name -> this should remain.
-        // host:port/image_name:v1 -> v1 should be replaced
-        if (lastColon > lastSlash) endIndex = lastColon
-        else endIndex = name.length()
+        if (tagValue.contains(':') || tagValue.contains('/')) {
+            // tag with ':' or '/' -> force use the tag value
+            return tagValue
+        } else {
+            // tag without ':' and '/' -> replace the tag part of original name
+            int lastColon = name.lastIndexOf(':')
+            int lastSlash = name.lastIndexOf('/')
 
-        return name.substring(0, endIndex) + ":" + tag
-    }
+            int endIndex;
+
+            // image_name -> this should remain
+            // host:port/image_name -> this should remain.
+            // host:port/image_name:v1 -> v1 should be replaced
+            if (lastColon > lastSlash) endIndex = lastColon
+            else endIndex = name.length()
 
-    private static String ucfirst(String str) {
-        StringBuffer sb = new StringBuffer(str);
-        sb.replace(0, 1, str.substring(0, 1).toUpperCase());
-        return sb.toString();
+            return name.substring(0, endIndex) + ":" + tagValue
+        }
     }
 
+    @Deprecated
```

Comment:
Is this supposed to be deprecated?

Reply:
Yes, `tags` DSL will be deprecated in future, we will no longer need this method.